api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java

```diff
@@ -778,6 +778,64 @@ public OrderType getOrderTypeByConceptClass(ConceptClass conceptClass) {
 		    "from OrderType where :conceptClass in elements(conceptClasses)").setParameter("conceptClass", conceptClass)
 		        .uniqueResult();
 	}
+
+	/**
+	 * @see org.openmrs.api.OrderService#getOrderTypesByClassName(String)
+	 */
+	@Override
+	public List<OrderType> getOrderTypesByClassName(String javaClassName) throws DAOException {
+		if (StringUtils.isBlank(javaClassName)) {
+			throw new APIException("javaClassName cannot be null");
+		}
+		
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<OrderType> cq = cb.createQuery(OrderType.class);
+		Root<OrderType> root = cq.from(OrderType.class);
+		
+		cq.where(cb.equal(root.get("javaClassName"), javaClassName));
+		
+		return session.createQuery(cq).getResultList();
+		
+	}
+
+	/**
+	 * @see org.openmrs.api.OrderService#getOrderTypesByClassName(String, boolean)
+	 */
+	@Override
+	public List<OrderType> getOrderTypesByClassName(String javaClassName, boolean includeSubclasses) throws DAOException {
+		if (!includeSubclasses) {
+			return getOrderTypesByClassName(javaClassName);
+		}
+		
+		if (StringUtils.isBlank(javaClassName)) {
+			throw new APIException("javaClassName cannot be null");
+		}
+
+		Class<?> superClass;
+		try {
+			superClass = Class.forName(javaClassName);
+		} catch (ClassNotFoundException e) {
+			throw new APIException("Invalid javaClassName: " + javaClassName, e);
+		}
+
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<OrderType> cq = cb.createQuery(OrderType.class);
+		Root<OrderType> root = cq.from(OrderType.class);
+		
+		List<OrderType> all = session.createQuery(cq.select(root)).getResultList();
```

Comment:
Per my comment above, there is already a DAO method to getOrderTypes, you should just use that.  You either need to also take in the "includeRetired" parameter, or always exclude retired I would think.